File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>s(nums.begin(), nums.end());

        int maxLen = 0;

        for(int n : s){
            if(s.find(n-1) == s.end()){
                int length = 1;
                while(s.find(n + length) != s.end()){
                    length++;
                }

                maxLen = max(maxLen, length);
            }
        }

        return maxLen;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        abc = set(nums)

        maxLen = 0

        for i in abc:
            first = i
            if (first - 1) not in abc:
                length = 1
                while (first + length) in abc:
                    length += 1
                maxLen = max(maxLen, length)
        
        return maxLen
